pkgs/applications/graphics/darktable/default.nix

```diff
@@ -18,32 +18,42 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ cmake ninja llvm pkgconfig intltool perl desktop-file-utils wrapGAppsHook ];
 
   buildInputs = [
-    cairo curl exiv2 glib gtk3 ilmbase lcms2 lensfun libX11 libexif
+    cairo curl exiv2 glib gtk3 ilmbase lcms2 lensfun libexif
     libgphoto2 libjpeg libpng librsvg libtiff openexr sqlite libxslt
     libsoup graphicsmagick json-glib openjpeg lua pugixml
-    colord colord-gtk libwebp libsecret gnome3.adwaita-icon-theme
-    osm-gps-map ocl-icd
-  ];
-    
+    libwebp libsecret gnome3.adwaita-icon-theme osm-gps-map pcre
+  ] ++ stdenv.lib.optionals stdenv.isLinux [
+    colord colord-gtk libX11 ocl-icd
+  ] ++ stdenv.lib.optional stdenv.isDarwin gtk-mac-integration;
+
   cmakeFlags = [
     "-DBUILD_USERMANUAL=False"
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
+    "-DUSE_COLORD=OFF"
+    "-DUSE_KWALLET=OFF"
   ];
 
+  ldLibraryPathPrefix = "$out/lib/darktable" + stdenv.lib.optionalString stdenv.isLinux ":${ocl-icd}/lib";
+
+  postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
+    mv $out/lib/darktable/libdarktable.dylib $out/lib
+  '';
+
   # darktable changed its rpath handling in commit
   # 83c70b876af6484506901e6b381304ae0d073d3c and as a result the
   # binaries can't find libdarktable.so, so change LD_LIBRARY_PATH in
   # the wrappers:
   preFixup = ''
     gappsWrapperArgs+=(
-      --prefix LD_LIBRARY_PATH ":" "$out/lib/darktable:${ocl-icd}/lib"
+      --prefix LD_LIBRARY_PATH ":" "${ldLibraryPathPrefix}"
```

Comment:
Do we need this on darwin? If so I'd expect it to be DYLD_LIBRARY_PATH instead.

Reply:
Fixed.